### PR TITLE
Add docker setup

### PR DESCRIPTION
**Pull request by ZorianaPro**
- [ ] build it in CI

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# pull official base image
+FROM node:13.12.0-alpine
+
+# set working directory
+WORKDIR /app
+
+# add `/app/node_modules/.bin` to $PATH
+ENV PATH /app/node_modules/.bin:$PATH
```
Comment (haensl) [inline Dockerfile:8]: Why do you need this?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+# install app dependencies
+COPY package.json ./
+COPY package-lock.json ./
+RUN npm install --silent
```
Comment (haensl) [inline Dockerfile:13]: I only recently learned that you should use [`npm ci`](https://docs.npmjs.com/cli/ci.html) to install dependencies for a production setup.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+RUN npm install react-scripts@3.4.0 -g --silent
```
Comment (haensl) [inline Dockerfile:14]: This should not be neccessary. Doesn't the project already use react-scripts/create-react-app?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+# add app
+COPY . ./
+# start app
+CMD ["npm", "start"]
```
Comment (haensl) [inline Dockerfile:19]: Docker distinguishes between `ENTRYPOINT` and `CMD`.
In order to make your container nice and flexible, you should use that.
--> entrypoint should be npm.
--> cmd should be the npm command you want to run.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+version: '3.4'
+
+services:
+  web:
+    build:
+      context: .
```
Comment (haensl) [inline docker-compose.yml:6]: You only need to specify build context or dockerfile if you want something different from the _default_. Looks like all defaults here.

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    command: npm start
+    image: zlesyk/hackathon-client:latest
+    container_name: hackathon-client
+    ports:
+      - 3000:3000
+    volumes:
+      - .:/app
+    depends_on:
+      - server
+  server:
```
Comment (haensl) [inline docker-compose.yml:20]: Does this setup (overall) work? I.e. can you locally start the container(s) and it works?

Comment (ZorianaPro) [reply docker-compose.yml:20]: it works)))

**Pull request merged**